frontend/src/app/main/ui/workspace/sidebar/options/menus/shadow.cljs

```diff
@@ -249,26 +240,36 @@
                           :on-open manage-on-open
                           :on-close manage-on-close}]]])]]))
 
+(def ^:private xf:add-index
+  (map-indexed (fn [index shadow]
+                 (assoc shadow ::index index))))
+
+
 (mf/defc shadow-menu
   {::mf/wrap-props false}
-  [props]
-  (let [ids            (unchecked-get props "ids")
-        type           (unchecked-get props "type")
-        values         (unchecked-get props "values")
+  [{:keys [ids type values]}]
 
-        shadows        (:shadow values [])
-        open-state-ref (mf/with-memo [] (l/atom {}))
-        has-shadows?   (or (= :multiple shadows) (some? (seq shadows)))
+  (let [shadows        (get values :shadow [])
+        shadows        (mf/with-memo [shadows]
+                         (if (= :multiple shadows)
+                           shadows
+                           (into [] xf:add-index shadows)))
 
-        state*         (mf/use-state {:show-content true
-                                      :disable-drag false})
+        open-state*    (mf/use-state {})
+        open-state     (deref open-state*)
 
-        state          (deref state*)
-        open?          (:show-content state)
-        disable-drag?  (:disable-drag state)
+        has-shadows?   (or (= :multiple shadows)
+                           (some? (seq shadows)))
+
```

Comment:
Can't this be memoized on `[shadows]`?

Reply:
Is already memoized, and think this (seq shadows) is also redundant, but I'm not going to change it on this PR